src/bc_protocol.rs

```diff
@@ -17,8 +17,9 @@ mod connection;
 mod media_packet;
 mod time;
 
-pub struct BcCamera {
-    address: SocketAddr,
+pub struct BcCamera<T: ToSocketAddrs> {
```

Comment:
I would prefer not to parameterize this struct. It's basically the runtime representation of the camera - and we don't have different kinds of cameras distinguished by the way they're named.

Instead, can you modify [`BcConnection::connect`](https://github.com/thirtythreeforty/neolink/blob/master/src/bc_protocol/connection.rs#L49) to accept a `ToSocketAddrs`? Then, you could pass the string straight in from the struct.

Since we create the socket with extra options, you'll still need to iterate over them in our connect_to helper. (This is [what `TcpStream::connect(ToSocketAddrs)` does](https://doc.rust-lang.org/src/std/net/tcp.rs.html#106), so that's good behavior.)

Reply:
Oh, also you'll need to remove [my previous attempt](https://github.com/thirtythreeforty/neolink/blob/32400971a532509aade690b01c84c04c757dfc1c/src/bc_protocol.rs#L84) at what you're implementing. The correct behavior is indeed to iterate over the results of the name resolution.

Reply:
I switched it up to just providing a standard new() with some sensible default values, but then it occurred to me that you probably meant to just construct a new BcConnection if BcConnection::connect() succeeds since we will destroy and reconstruct a new BcConnection on failure anyways.